nalgebra-sparse/tests/unit_tests/coo.rs

```diff
@@ -252,3 +252,97 @@ fn coo_push_out_of_bounds_entries() {
         assert_panics!(coo.clone().push(3, 2, 1));
     }
 }
+
+#[test]
+fn coo_push_mat_valid_entries() {
+    let mut coo = CooMatrix::new(3, 3);
+
+    // Works with static
+    {
+        // new is row-major...
+        let inserted = nalgebra::SMatrix::<i32, 2, 2>::new(1, 2, 3, 4);
+        coo.push_mat(1, 1, &inserted);
+
+        // insert happens column-major, so expect transposition when read this way
+        assert_eq!(
+            coo.triplet_iter().collect::<Vec<_>>(),
+            vec![(1, 1, &1), (2, 1, &3), (1, 2, &2), (2, 2, &4)]
+        );
+    }
+
+    // Works with owned dynamic
+    {
+        let inserted = nalgebra::DMatrix::<i32>::repeat(1, 2, 5);
+        coo.push_mat(0, 0, &inserted);
+
+        assert_eq!(
+            coo.triplet_iter().collect::<Vec<_>>(),
+            vec![
+                (1, 1, &1),
+                (2, 1, &3),
+                (1, 2, &2),
+                (2, 2, &4),
+                (0, 0, &5),
+                (0, 1, &5)
+            ]
+        );
+    }
+
+    // Works with sliced
+    {
+        let source = nalgebra::SMatrix::<i32, 2, 2>::new(6, 7, 8, 9);
+        let sliced = source.fixed_slice::<2, 1>(0, 0);
+        coo.push_mat(1, 0, &sliced);
+
+        assert_eq!(
+            coo.triplet_iter().collect::<Vec<_>>(),
+            vec![
+                (1, 1, &1),
+                (2, 1, &3),
+                (1, 2, &2),
+                (2, 2, &4),
+                (0, 0, &5),
+                (0, 1, &5),
+                (1, 0, &6),
+                (2, 0, &8)
+            ]
+        );
+    }
+
+    let csr = nalgebra_sparse::csr::CsrMatrix::from(&coo);
```

Comment:
axe this

Reply:
https://github.com/dimforge/nalgebra/pull/904/commits/6b1b9077fe76a2644501f07179a2d2e7823821ee